### ISSUE TITLE
Reported: Open Broadcaster Software + macOS

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#282]
A user said Graillon was crashing this host.

**Comment by p0nce** [comment]
Reproduced with LDC 1.8 + a plugin + macOS 10.13.2 + VST + 64-bit

**Comment by p0nce** [comment]
A script to reproduce on macOS with the Distort or Clip it example.
```sh
#!/bin/sh

# build the plug-in
dplug-build --compiler ldc --config VST -a x86_64

# copy to standard VST directory
sudo cp -R builds/macOS-64-bit-VST/*.vst /Library/Audio/Plug-Ins/VST

# OBS need to be run in its resource directory
cd /Applications/OBS.app/Contents/Resources/bin/

# launch OBS attached
lldb obs
```

**Comment by p0nce** [comment]
Problem is `effGetEffectName` being called with a null `_client`

**Comment by p0nce** [comment]
Looks like a codegen problem but isn't since DMD has a similar problem.

**Comment by p0nce** [comment]
For some reason that host erase our `_effect.user` field, maybe find another way to pass the VSTClient pointer

**Comment by p0nce** [comment]
Fixed in 69900a9667362fc421c5b52b620595a2d0fe6bec
